internal/pkg/state/local/manager.go

```diff
@@ -272,6 +273,9 @@ func (u *UnitOfWork) Invoke() error {
 
 // workersFor each level (branches, configs, rows).
 func (u *UnitOfWork) workersFor(level int) *Workers {
+	u.lock.Lock()
+	defer u.lock.Unlock()
```

Comment:
Tu chybalo volanie zamku, metoda sa moze volat paralelne (a problemy to zacalo sposobovat az po novych zmenach).